src/app/services/api/amo-api/amo-api.service.ts

```diff
@@ -0,0 +1,88 @@
+import { AmoApiModule, POST_MESSAGE_HTTP_CLIENT_TOKEN } from './amo-api.module';
+import { domainSelector } from '../../../store/crm-context/selectors';
+import { HttpClient } from '@angular/common/http';
+import { Inject, Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { accessTokenSelector } from 'src/app/store/access-token/selectors';
+import { environment } from 'src/environments/environment';
+import { ApiService } from '../api.service';
+import { getEntityCustomFieldsAllEndpoint } from './endpoints/get-entity-custom-fields-all.endpoint';
+import { Observable, map } from 'rxjs';
+import { getLeadEndpoint } from './endpoints/get-lead.endpoint';
+import { ICrmCustomFieldValues } from 'src/app/interfaces/crm-custom-field-value.interface';
+import { ICrmCustomFieldInfo } from 'src/app/interfaces/crm-custom-field.interface-info';
+import { patchLeadEndpoint } from './endpoints/patch-lead.endpoint';
+
+const BASE_URL_COMPILER: (domain: string) => string = environment.getAmoBaseUrl;
+
+@Injectable({
+  providedIn: AmoApiModule,
```

Comment:
Изначально указывал провайды в модулях, но потом наткнулся на примечание в доках, что будто бы предпочтительнее делать это в декораторе. Написано (https://angular.io/guide/providers) следующее:

``` JavaScript
import { [Injectable](https://angular.io/api/core/Injectable) } from '@angular/core';
import { UserModule } from './user.module';

@[Injectable](https://angular.io/api/core/Injectable)({
  providedIn: UserModule,
})
export class UserService {
}
```
"The example above shows the preferred way to provide a service in a module. This method is preferred because it enables tree-shaking of the service if nothing injects it. ..."